salt/engines/ircbot.py

```diff
@@ -94,14 +94,15 @@ def __init__(self, nick, host, port=6667, username=None, password=None, channels
         self.allow_nicks = allow_nicks
         self.disable_query = disable_query
         self.io_loop = tornado.ioloop.IOLoop(make_current=False)
+        self.io_loop.make_current()
```

Comment:
I'm not familiar with `make_current`, but is there a reason why we are instantiating the `IOLoop` instance and then starting it as a separate action?

Reply:
if `make_current` is `True` and there is one current IOLoop it raises an exception. But here we want to have an own one because it's an engine that is an independent process.
I should probably have a comment for this in the code.

Reply:
Yes, that would be helpful for future people that maintain this code, to prevent regressions. Thanks for the explanation!